### PR TITLE
Add virtual package for x86 CPU features

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
@cla-bot check

**Comment by msarahan** [comment]
@cla-bot check

**Comment by msarahan** [comment]
We have to manually add signers after we verify receipt of the e-sign thing.  It's annoying, but it's what our legal team said we needed.  You're all good now on any of the conda org repos, though.

**Comment by mbargull** [comment]
(cc @johanneskoester: This is what I mentioned last Thursday wrt. custom builds for HPC environments.)

File: conda/cli/main_info.py [hunk 1]
```diff
@@ -239,6 +240,14 @@ def get_main_info_str(info_dict):
         '%s=%s' % tuple(x) for x in info_dict['virtual_pkgs']])
     info_dict['_rtwro'] = ('writable' if info_dict['root_writable'] else 'read only')
 
+    cpu_flags_lines = []
+    for flag in info_dict['cpu_flags']:
+        if cpu_flags_lines and len(cpu_flags_lines[-1]) + len(flag) + 1 < 100:
+            cpu_flags_lines[-1] += " " + flag
+        else:
+            cpu_flags_lines.append(flag)
+    info_dict['_cpu_flags'] = ('\n' + 26 * ' ').join(line for line in cpu_flags_lines)
```
Comment (mbargull) [inline conda/cli/main_info.py]: It might make sense to only output this when `config --all` is run.

File: conda/core/index.py [hunk 2]
```diff
@@ -171,6 +171,19 @@ def _supplement_index_with_system(index):
         rec = _make_virtual_package('__' + libc_family, libc_version)
         index[rec] = rec
 
+    for flag in context.cpu_flags:
+        if os.getenv("CONDA_OVERRIDE_CPU_FEATURE_{}".format(flag.upper()), "1") == "":
+            continue
+        rec = _make_virtual_package('__cpu_feature_' + flag.lower(), "1.0.0")
```
Comment (mbargull) [inline conda/core/index.py]:
```suggestion
        rec = _make_virtual_package('__cpu_feature_' + flag.lower(), "1")
```
`1` is a valid "version", too and having minor/patch parts does not make much sense for flags.

File: conda/core/index.py [hunk 3]
```diff
@@ -171,6 +171,19 @@ def _supplement_index_with_system(index):
         rec = _make_virtual_package('__' + libc_family, libc_version)
         index[rec] = rec
 
+    for flag in context.cpu_flags:
```
Comment (mbargull) [inline conda/core/index.py]: Do we need a whitelist or name mapping potentially?
I don't know whether there are any conflicting flag names between different CPU vendors/architectures, e.g., whether feature `xyz` exists for Intel/AMD x86 but also for ARM/PPC but with a different meaning.

File: conda/core/index.py (hunk 2 above)
Comment (mbargull) [reply conda/core/index.py]: I'd also like us to have an option to disable/remove such flag, e.g., to ensure no feature-dependent packages are used during build time.
(That doesn't have to be addressed in this PR though. I'm generally not too happy with those `OVERRIDE` env vars and will open an issue addressing that.)

File: conda/core/index.py (hunk 3 above)
Comment (isuruf) [reply conda/core/index.py]: If they have a different meaning, so be it. I don't see any reason to not allow conflicting feature names if they are in different subdirs.

**Comment by isuruf** [comment]
See https://github.com/conda/conda/pull/9930 for a different approach.

**Comment by travishathaway** [comment]
This pull request has become stale and is therefore being closed. Please re-open if you would like to continue development on this.

**Pull request closed without merging**